pyiron_atomistics/atomistics/structure/neighbors.py

```diff
@@ -218,6 +218,9 @@ def _get_distances_and_indices(
             distance_upper_bound=cutoff_radius,
             p=self.norm_order,
         )
+        shape = np.asarray(positions).shape[:-1]+(num_neighbors,)
```

Comment:
Converting to positions to a numpy array is potentially memory intensive (if it was a list before) just to get the shape. Since `_get_wrapped_positions` returns an array already (where this conversion is actually useful), I'd do something like
```python
positions = self._get_wrapped_positions(positions)
... = self._tree.query(positions, ...)
shape = positions.shape[:-1]+(num_neighbors,)
```